common/components/WalletDecrypt/WalletDecrypt.tsx

```diff
@@ -259,21 +262,30 @@ export class WalletDecrypt extends Component<Props, State> {
           {!selectedWallet.isReadOnly && 'Unlock your'} {translate(selectedWallet.lid)}
         </h2>
         <section className="WalletDecrypt-decrypt-form">
-          <selectedWallet.component
-            value={this.state.value}
-            onChange={this.onChange}
-            onUnlock={this.onUnlock}
-            showNotification={this.props.showNotification}
-            isWalletPending={
-              this.state.selectedWalletKey === InsecureWalletName.KEYSTORE_FILE
-                ? this.props.isWalletPending
-                : undefined
-            }
-            isPasswordPending={
-              this.state.selectedWalletKey === InsecureWalletName.KEYSTORE_FILE
-                ? this.props.isPasswordPending
-                : undefined
-            }
+          <Route
+            render={routerProps => (
+              <selectedWallet.component
+                value={this.state.value}
+                onChange={this.onChange}
+                onUnlock={value => {
+                  if (selectedWallet.redirect) {
+                    routerProps.history.push(selectedWallet.redirect);
+                  }
+                  this.onUnlock(value);
+                }}
+                showNotification={this.props.showNotification}
+                isWalletPending={
+                  this.state.selectedWalletKey === InsecureWalletName.KEYSTORE_FILE
+                    ? this.props.isWalletPending
+                    : undefined
+                }
+                isPasswordPending={
+                  this.state.selectedWalletKey === InsecureWalletName.KEYSTORE_FILE
+                    ? this.props.isPasswordPending
+                    : undefined
+                }
+              />
+            )}
```

Comment:
You can't redirect any unlock to /account/send, because the wallet may be unlocked on a different page. Also from a separation of concerns standpoint, I don't like the idea of the standalone `<WalletDecrypt/>` component being responsible for /account's behavior.

This should most likely be handled somewhere in the `containers/SendTransaction` domain. Perhaps the Send form should do the redirect, if the wallet it receives is read-only.